engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -80,24 +80,36 @@ def prepare_scc_announce_request(uri_path, auth, params)
       # to SCC.
       # SCC will make sure to handle the data correctly. This removes the need
       # to adapt here if information send by the client changes.
-      scc_request.body = {
+      scc_req_body = {
         hostname: params['hostname'],
         hwinfo: params['hwinfo'],
-        byos: true
-      }.to_json
+        byos_mode: params['proxy_byos_mode']
+      }
+      # when system is payg, we do not know whether it's hybrid or not
+      # we send the login and password information to skip the validation
+      # on the SCC side, that info is enough to validate the product later on
+      # if the system is, in fact, hybrid
+      if params['proxy_byos_mode'].casecmp('hybrid').zero?
```

Comment:
Let's make clear why we add the system's credentials here. Also, I don't understand why we need to compare case insensitive here?

```suggestion
      # When activating a BYOS extension on top of a PAYG system ('hybrid mode'), the system already has credentials in RMT, but is not known to SCC. We announce it to SCC including its credentials in this case.
      if params['proxy_byos_mode'] == 'hybrid'
```

In general, wouldn't it be useful to provide the @system instance to all of the methods in this file, that would safe us from passing around all those parameters.